### PR TITLE
reallocarray: if we are asked to allocate 0, return malloc(1)

### PR DESCRIPTION
**Pull request by rminnich**
This ensures that, now and into the future, reallocarracy
of 0 members or zero size will return a non-nil pointer.

We either do this or change the meaning or use of reallocarray.
Currently, the slice code requires a non-nil return.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by elbing** [comment]
You rule. @dancrossync?

**Comment by sevki** [comment]
LGTM

File: sys/src/libc/port/reallocarray.c [hunk 1]
```diff
@@ -41,7 +41,7 @@ reallocarray(void *optr, size_t nmemb, size_t size)
 	 */
 	if (nmemb == 0 || size == 0) {
 		free(optr);
-		return malloc(0);
+		return malloc(1);
```
Comment (dancrossnyc) [inline sys/src/libc/port/reallocarray.c]: Malloc(0) already returns a valid pointer. What was the motivation for this change?

**Comment by rminnich** [comment]
it was not returning a valid pointer. It was returning NULL in the kernel, causing slice code to panic.

**Comment by dancrossnyc** [comment]
oh bummer. ok then. :-(

**Pull request merged**